controllers/quay/quayregistry_controller.go

```diff
@@ -120,10 +120,84 @@ func (r *QuayRegistryReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, nil
 	}
 
-	created := v1.GetCondition(quay.Status.Conditions, v1.ConditionComponentsCreated)
-	if created != nil && created.Reason == v1.ConditionReasonMigrationsInProgress {
-		log.Info("migrations in progress, skipping reconcile")
-		return ctrl.Result{}, nil
+	created := v1.GetCondition(updatedQuay.Status.Conditions, v1.ConditionComponentsCreated)
+	migrationInProgress := created != nil && created.Reason == v1.ConditionReasonMigrationsInProgress
+	if migrationInProgress {
+		log.Info("++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
+		log.Info("migration currently in progress!")
+		log.Info(fmt.Sprintf("currentVersion is: '%s'", updatedQuay.Status.CurrentVersion))
+
+		err := wait.Poll(upgradePollInterval, upgradePollTimeout, func() (bool, error) {
```

Comment:
I don't think we need this `Poll`. We can check only once and if the migration is running return a reschedule, we will check again during the next cycle.

Reply:
true!

Reply:
done